internal/indexcoord/index_builder.go

```diff
@@ -154,6 +155,48 @@ func (ib *indexBuilder) schedule() {
 	}
 }
 
+func (ib *indexBuilder) toIndexBuildRequest(indexMeta *indexpb.IndexMeta) (*indexpb.CreateIndexRequest, error) {
+	// 1. populate: segment info from datacoord
+	segID := indexMeta.Req.SegmentID
+	segmentInfos, err := ib.ic.broker.getDataSegmentInfosByIDs([]UniqueID{segID})
+	if err != nil {
+		log.Error("get SegmentInfo failed", zap.Error(err))
+		return nil, fmt.Errorf("get SegmentInfo failed for segmentID %d", segID)
+	}
+
+	// 2. populate: field schema from rootcoord
+	collID := indexMeta.Req.CollectionID
```

Comment:
for old req, there are no collectionID and FieldIDs, it can not be compatible

Reply:
fixed